### ISSUE TITLE
Inner SatisfactionLink is not being evaluated

### ISSUE THREAD

**Issue by vsbogd** [opencog/atomspace#2221]
Following minimal example returns empty set:
```scm
(Inheritance (Concept "mouse") (Concept "animal"))
(Inheritance (Concept "mouse") (Concept "mammal"))

(cog-execute! (GetLink (Variable "x") 
  (And 
    (Inheritance (Variable "x") (Concept "animal"))
    (Satisfaction (Variable "y") (And (Inheritance (Variable "x") (Variable "y"))) ) 
  )
))
```

While standalone `SatisfactionLink` returns true:
```scm
(cog-evaluate! (Satisfaction (Variable "y") (And (Inheritance (Concept "mouse") (Variable "y")) )) )
```

The reason is that `SatisfactionLink` is being matched instead of being evaluated.

**Comment by linas** [comment]
Fixed by #2224

**Comment by vsbogd** [comment]
It is fixed, thanks!
